File: film_randomized_mobile/lib/services/auth_api.dart
import 'api_client.dart';

class AuthApi {
  AuthApi._();

  static Future<Map<String, dynamic>> login(
      String username, String password) async {
    return ApiClient().post('/auth/login', {
      'username': username,
      'password': password,
    });
  }

  static Future<Map<String, dynamic>> register(
      String username, String email, String password) async {
    return ApiClient().post('/auth/register', {
      'username': username,
      'email': email,
      'password': password,
    });
  }

  static Future<Map<String, dynamic>> me(String token) async {
    return ApiClient().get('/auth/me');
  }

  static Future<Map<String, dynamic>> forgotPassword(String email) async {
    return ApiClient().post('/auth/forgot-password', {'email': email});
  }

  static Future<Map<String, dynamic>> resetPassword(
      String token, String newPassword) async {
    return ApiClient().post('/auth/reset-password', {
      'token': token,
      'newPassword': newPassword,
    });
  }

  static Future<Map<String, dynamic>> changePassword(
      String currentPassword, String newPassword, String token) async {
    return ApiClient().post('/auth/change-password', {
      'currentPassword': currentPassword,
      'newPassword': newPassword,
    });
  }
}

File: film-randomized/src/shared/services/authApi.ts
import { apiClient } from './apiClient.js';
import { AuthResponse, User } from '../types/index.js';

export const authApi = {
  async login(username: string, password: string): Promise<AuthResponse> {
    return apiClient.post<AuthResponse>('/auth/login', { username, password });
  },

  async register(username: string, email: string, password: string): Promise<AuthResponse> {
    return apiClient.post<AuthResponse>('/auth/register', { username, email, password });
  },

  async getCurrentUser(token: string | null): Promise<User> {
    return apiClient.get<User>('/auth/me', token);
  },

  async forgotPassword(email: string): Promise<unknown> {
    return apiClient.post<unknown>('/auth/forgot-password', { email });
  },

  async resetPassword(token: string, newPassword: string): Promise<unknown> {
    return apiClient.post<unknown>('/auth/reset-password', { token, newPassword });
  },

  async changePassword(
    currentPassword: string,
    newPassword: string,
    token: string | null,
  ): Promise<unknown> {
    return apiClient.post<unknown>(
      '/auth/change-password',
      { currentPassword, newPassword },
      token,
    );
  },
};
